ouroboros-consensus/test-storage/Test/Ouroboros/Storage/TestBlock.hs

```diff
@@ -0,0 +1,384 @@
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE NamedFieldPuns             #-}
+{-# LANGUAGE RecordWildCards            #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE TupleSections              #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE UndecidableInstances       #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Test.Ouroboros.Storage.TestBlock where
+
+import qualified Codec.CBOR.Read as CBOR
+import qualified Codec.CBOR.Write as CBOR
+import           Codec.Serialise (Serialise (decode, encode))
+import           Control.Monad (forM)
+import           Control.Monad.Except (throwError)
+import           Data.ByteString.Builder (Builder)
+import qualified Data.ByteString.Lazy as Lazy
+import           Data.FingerTree.Strict (Measured (..))
+import           Data.Functor (($>))
+import           Data.Hashable
+import           Data.List.NonEmpty (NonEmpty)
+import qualified Data.List.NonEmpty as NE
+import qualified Data.Map.Strict as Map
+import           Data.Maybe (maybeToList)
+import           Data.Word
+import           GHC.Generics (Generic)
+import           GHC.Stack (HasCallStack)
+import           Test.QuickCheck
+
+import           Control.Monad.Class.MonadThrow
+
+import           Cardano.Crypto.DSIGN
+import           Cardano.Prelude (NoUnexpectedThunks)
+
+import           Ouroboros.Network.Block
+import           Ouroboros.Network.MockChain.Chain (Point)
+import qualified Ouroboros.Network.MockChain.Chain as Chain
+
+import           Ouroboros.Consensus.Block
+import           Ouroboros.Consensus.Ledger.Abstract
+import           Ouroboros.Consensus.Ledger.Extended
+import           Ouroboros.Consensus.NodeId (NodeId (..))
+import           Ouroboros.Consensus.Protocol.Abstract
+import           Ouroboros.Consensus.Protocol.BFT
+import           Ouroboros.Consensus.Protocol.Signed
+import           Ouroboros.Consensus.Util.Condense
+import           Ouroboros.Consensus.Util.Orphans ()
+import qualified Ouroboros.Consensus.Util.SlotBounded as SB
+
+import           Ouroboros.Storage.FS.API (HasFS (..), withFile)
+import           Ouroboros.Storage.FS.API.Types
+
+import           Test.Util.Orphans.Arbitrary ()
+
+{-------------------------------------------------------------------------------
+  TestBlock
+-------------------------------------------------------------------------------}
+
+data TestBlock = TestBlock {
+      testHeader :: !TestHeader
+    , testBody   :: !TestBody
+    }
+  deriving stock    (Show, Eq, Generic)
+  deriving anyclass (NoUnexpectedThunks, Serialise)
```

Comment:
Might be good to run this with the no-unexpected-thunks assertion enabled.

Reply:
The tests passed with the flag on.